### ISSUE TITLE
How to check if Plots are broken?

### ISSUE THREAD

**Issue by TschiAR** [Chia-Network/chia-blockchain#1963]
Hey there,

i now have about 26 TB Plots ( K33/K32). LAST XCH WIN: Confirmed at height 89558
12 days since then..NOTHING!!!

I ask myself now if my plots are borken?

Is there a posibility to check it somehow on windows system? 

i need a step by step tutorial please...

Thanks a lot for your help.

**Comment by eFishCent** [comment]
You can use Powershell command line interface (CLI) and run `chia plots check`. Instructions can be found here: https://github.com/Chia-Network/chia-blockchain/wiki/CLI-Commands-Reference

A bit of google learning is required to learn how to use Powershell 😄 

Additional useful Windows tips can be found here: https://github.com/Chia-Network/chia-blockchain/wiki/Windows-Tips-and-Tricks

**Comment by flotti455** [comment]
Last time I needed to rely on checking for broken plots (~4 weeks ago, so older Chia version), `chia plots check` completely failed to report on plots that were broken - it happily reported "x of y proofs" found, despite the plots being indeed broken:

The broken plots were plots whose transfer to the farmer were interrupted due to a crashing plotter computer. Nevertheless, file says indicated a healthy k32 file size (Windows file copy shenanigans, I guess).

Only by checksumming the files to the original files, I could spot broken plots. A further check on byte-by-byte comparison showed nulled bytes filling up the k32 file size.

**Comment by TschiAR** [comment]
REALLY

I understand nothing... all the plotting and chia support for month for this? i need a step by step tutorial?

there has to be a more easy way to check that? i guess there are thousands of broken plots wasting space....

**Comment by eFishCent** [comment]
tldr; There is no 100% way to verify a plot, this is just the way it is due to math, probabilities and crypto.

The way plots are created and checked there is no way to 100% verify a plot without going through every possible permutation of blockchain challenge in the universe. If you suspect there is a possible problem with your plots due to bad hardware or copying errors, you should run `chia plots check -n 1000` The higher number of random challenges will cover more parts of the file, but again, this is not 100%.

As long as there are some valid proofs in the plot, it will still work. It just won't work as well as a plot that has no errors in it.

@TschiAR - Chia is still an early technology. There isn't step by step tutorial for every situation. You're going to need to learn a bit on your own. The information link I provided is the best you can find at the moment.

FYI - I am a community contributor and not affiliated with Chia Networks at all.

**Comment by flotti455** [comment]
> There is no 100% way to verify a plot, this is just the way math, probabilities and crypto work.

Well, that's where _engineering_ comes to the rescue: make plausibility checks such as "does the file follow the Chia plot file format definition" (magic values, expected metadata thoughout the whole plot).

With files so big, this is a low-hanging fruit for improvement - a plot with tens-of-gigabytes of garbage shall not happily pass the filter.

**Comment by eFishCent** [comment]
> Well, that's where _engineering_ comes to the rescue: make plausibility checks such as "does the file follow the Chia plot file format definition" (magic values, expected metadata thoughout the whole plot, etc.).

Would love you to solve it. Pull requests welcome :smile:
